Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -2161,10 +2161,46 @@ def test_url_requests(self):
 
 
 class TestHelper(unittest.TestCase):
+    def mock_interactive_session(self, inputs):
+        """
+        Given a list of inputs, run an interactive help session.  Returns a string
+        of what would be shown on screen.
+        """
+        input_iter = iter(inputs)
+
+        def mock_getline(prompt):
+            output.write(prompt)
+            next_input = next(input_iter)
+            output.write(next_input + os.linesep)
+            return next_input
+
+        with captured_stdout() as output:
+            helper = pydoc.Helper(output=output)
+            with unittest.mock.patch.object(helper, "getline", mock_getline):
+                helper.interact()
+
+        return output.getvalue().replace(os.linesep, "\n")
```

Comment:
Perhaps having it return the array of lines might be nicer?
```suggestion
        return output.getvalue().split(os.linesep)
```

Reply:
will check thanks